### PR TITLE
Documentation tool updates and st_archive fix

### LINKED ISSUE #594: problem with st_archive and short runs (< 1 month)
If the short term archiver is run prior to the end of month boundary then it errors out because it can't find the component monthly history file corresponding to the component restart history file. 

The st_archive log errors from a F-WACCM case are:

==> f.e15.FWscAMIP.f19_f19.misc08_cam5_4_81.002.st_archive.570708 <==
out=
err=ncdump: nhfil: No such variable
histfiles_savein_rundir ['f.e15.FWscAMIP.f19_f19.misc08_cam5_4_81.002.cam.h0.1979-01.nc', 'f.e15.FWscAMIP.f19_f19.misc08_cam5_4_81.002.cam.h1.1979-01-01-00000.nc', 'f.e15.FWscAMIP.f19_f19.misc08_cam5_4_81.002.cam.h2.1979-01-01-00000.nc'] 
doing short term archiving for clm (lnd)
removing interim restart file /glade/scratch/mmills/f.e15.FWscAMIP.f19_f19.misc08_cam5_4_81.002/run/f.e15.FWscAMIP.f19_f19.misc08_cam5_4_81.002.clm2.r.1979-01-06-00000.nc
 WARNING: ncdump -v locfnh /glade/scratch/mmills/f.e15.FWscAMIP.f19_f19.misc08_cam5_4_81.002/run/f.e15.FWscAMIP.f19_f19.misc08_cam5_4_81.002.clm2.rh0.1979-01-06-00000.nc  failed rc=1
out=
err=ncdump: locfnh: No such variable
removing interim restart file /glade/scratch/mmills/f.e15.FWscAMIP.f19_f19.misc08_cam5_4_81.002/run/f.e15.FWscAMIP.f19_f19.misc08_cam5_4_81.002.clm2.rh0.1979-01-06-00000.nc
ERROR: restart file /glade/scratch/mmills/f.e15.FWscAMIP.f19_f19.misc08_cam5_4_81.002/run/f.e15.FWscAMIP.f19_f19.misc08_cam5_4_81.002.clm2.h0.1979-01.nc does not exist

### LINKED ISSUE #1715: Remove src/drivers/mct/doc
It looks like the material in src/drivers/mct/doc is out of data and replaced by doc/source/driver_cpl.  So lets remove it.

### LINKED ISSUE #1731: multi-instance rpointer file names in config_archive.xml don't match $rundir
config_archive.xml:
      <rpointer_file>rpointer$NINST_STRING.atm</rpointer_file>'
rundir:
     rpointer.atm_$NINST_STRING

Similarly for all other (multi-instance) components.
I vote for changing config_archive.xml, so that the instance
number is always attached to the component name, 
rather than the file type.   This would also match the
restart, initial, and history files.

Thanks
Kevin

### PR DESCRIPTION
**Pull request by bertinia**
Includes many commits for XML 2 html auto-generated documentation. 
Fix to st_archive for multi-instance. 
Remove old coupler documentation. 

Test suite: b_N3.B1850.f19_g17 with interim restarts saved to check rpointer filenames and content for correctness. Also tested a short run with change to config_archive.xml for fix to #594
Test baseline: N/A
Test namelist changes: N/A
Test status: bit for bit

Fixes [CIME Github issue #] 
Fixes #1731  
Fixes #1715 
Fixes #594

User interface changes?: none

Update gh-pages html (Y/N)?: Y

Code review:

### CODE REVIEW

**Comment by billsacks** [comment]
@bertinia Issues will only auto-close if you put the string "fixes" separately before each issue:
Fixes issue1
Fixes issue2
etc.

**Comment by rljacob** [comment]
You need to add labels too.

File: scripts/lib/CIME/XML/component.py [hunk 1]
```diff
@@ -284,3 +284,4 @@ def print_values(self):
         for v in sorted(compsets.iteritems()):
             label, definition = v
             logger.info("   {:20s} : {}".format(label, definition))
+
```
Comment (jedwards4b) [inline scripts/lib/CIME/XML/component.py]: Remove this change.

File: src/components/stub_comps/srof/cime_config/config_component.xml [hunk 2]
```diff
@@ -4,6 +4,10 @@
 
 <entry_id>
 
+  <description mustmatch="SROF">
```
Comment (jedwards4b) [inline src/components/stub_comps/srof/cime_config/config_component.xml]: Please make this consistent with the v3 format.

File: src/components/stub_comps/swav/cime_config/config_component.xml [hunk 3]
```diff
@@ -8,6 +8,10 @@
     <desc wav="SWAV">Stub wave component</desc>
   </description>
 
+  <description>
```
Comment (jedwards4b) [inline src/components/stub_comps/swav/cime_config/config_component.xml]: use v3 format

File: config/xml_schemas/entry_id_namelist_old.xsd [hunk 4]
```diff
@@ -0,0 +1,33 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
[the remark this answers is not in the document]
Comment (bertinia) [reply config/xml_schemas/entry_id_namelist_old.xsd]: removed.

File: scripts/lib/CIME/XML/grids.py [hunk 5]
```diff
@@ -520,6 +520,8 @@ def _print_values_v2(self, long_output=None):
                     if gridname != "null":
                         logger.info ("    {}".format(domains[gridname]))
 
+
```
[the remark this answers is not in the document]
Comment (bertinia) [reply scripts/lib/CIME/XML/grids.py]: remove what?

[the remark this answers is not in the document]
Comment (jedwards4b) [reply scripts/lib/CIME/XML/grids.py]: extra empty line

File: scripts/lib/CIME/case.py [hunk 6]
```diff
@@ -458,6 +458,54 @@ def _set_compset(self, compset_name, files):
         if compset_alias is None:
             logger.info("Did not find a compset match for longname {} ".format(compset_name))
             self._compsetname = compset_name
+            files = Files()
```
[the remark this answers is not in the document]
Comment (bertinia) [reply scripts/lib/CIME/case.py]: old version that was in my fork - updated to master.

File: config/cesm/config_archive.xml [hunk 7]
```diff
@@ -2,11 +2,12 @@
```
[the remark this answers is not in the document]
Comment (jedwards4b) [reply config/cesm/config_archive.xml]: this was incorrect.

File: scripts/query_config [hunk 8]
```diff
@@ -73,7 +73,9 @@ def query_compsets(name):
     else:
         print_compset(name, files)
 
+###############################################################################
```
[the remark this answers is not in the document]
Comment (bertinia) [reply scripts/query_config]: removed from query_config but this unnecessary decoration is in a lot of the scripts including create_newcase. We should be using docstrings instead for our in-line comments so the sphinx autodoc extension picks up the comment. I'll try and clean these up as part of issue #1668.

File: config/cesm/config_archive.xml (hunk 7 above)
[the remark this answers is not in the document]
Comment (bertinia) [reply config/cesm/config_archive.xml]: This file is still required but reverted it to the master in favor of PR #1830

File: scripts/lib/CIME/XML/component.py (hunk 1 above)
Comment (bertinia) [reply scripts/lib/CIME/XML/component.py]: fixed.

File: src/components/stub_comps/srof/cime_config/config_component.xml (hunk 2 above)
Comment (bertinia) [reply src/components/stub_comps/srof/cime_config/config_component.xml]: fixed.

File: src/components/stub_comps/swav/cime_config/config_component.xml (hunk 3 above)
Comment (bertinia) [reply src/components/stub_comps/swav/cime_config/config_component.xml]: fixed.

**Pull request merged**